src/Geometry/GRectangle.class.st

```diff
@@ -28,8 +28,6 @@ GRectangle class >> origin: point1 corner: point2 [
 	or := (point1 x min: point2 x) , (point1 y min: point2 y).
 	cor := (point1 x max: point2 x) , (point1 y max: point2 y).
 	
-	or = cor ifTrue: [ self error: 'This is not a rectangle but a point.' ].
```

Comment:
In which case do you want a rectangle that is a point?

Reply:
I've checked and from the mathematical point of view a point cannot be considered as a rectangle.

Reply:
What is your reference? From my perspective, users can have empty shapes like an empty ellipse or empty rectangles, or empty polygons.

Reply:
I checked multiple sources such as Wikipedia, Wolfram, the dictionary...

All of them agree on "quadrilateral with opposite sides of equal lengths a and b, and with four right angles"

A point cannot have right angles. Thus, a point is not a rectangle.

Reply:
For empty shapes, I would create a NullShape. For intersections, for example, it would always return an empty collection.

Or if your empty shape acts as a point, then I would use a point instead. IIRC points can react to #intersections:

Reply:
I will use points for that then.